src/js/DatePicker.js

```diff
@@ -0,0 +1,436 @@
+const $ = window.$ || require("jquery");
+require('jquery-ui/ui/widgets/datepicker');
+
+class DatePicker {
+    /**
+     * DatePicker
+     * @param {DatePickerUtils} utils
+     * @param {DatePickerHighlightUtils} highlightUtils
+     * @param {DatePickerAccessibleLabelUtils} accessibleLabelUtils
+     * @param {DatePickerNavigationManager} navigationManager
+     */
+    constructor (utils, highlightUtils, accessibleLabelUtils, navigationManager) {
+        /**
+         * Root html element
+         * @type {JQuery|null}
+         */
+        this.$html = null;
+        /**
+         * Calendar trigger button
+         * @type {JQuery|null}
+         */
+        this.$triggerButton = null;
+        /**
+         * Date input
+         * @type {JQuery|null}
+         */
+        this.$dateInput = null;
+        /**
+         * Calendar next month link
+         * @type {JQuery|null}
+         */
+        this.$nextLink = null;
+        /**
+         * Calendar previous month link
+         * @type {JQuery|null}
+         */
+        this.$prevLink = null;
+        /**
+         * Calendar today link
+         * @type {JQuery|null}
+         */
+        this.$todayLink = null;
+        /**
+         * Calendar wrapper
+         * @type {JQuery|null}
+         */
+        this.$datePickerContainer = null;
```

Comment:
Have you seen the JS class properties?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Class_fields

Babel here: https://babeljs.io/docs/en/babel-plugin-transform-class-properties/